examples/cpp/custom_payload_pool/PublisherApp.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file PublisherApp.cpp
+ *
+ */
+
+#include "PublisherApp.hpp"
+
+#include <csignal>
+#include <thread>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::rtps;
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace custom_payload_pool {
+
+PublisherApp::PublisherApp(
+        const CLIParser::publisher_config& config,
+        const std::string& topic_name)
+    : participant_(nullptr)
+    , publisher_(nullptr)
+    , topic_(nullptr)
+    , writer_(nullptr)
+    , type_(new CustomPayloadPoolDataPubSubType())
+    , matched_(0)
+    , samples_(config.samples)
+    , stop_(false)
+    , period_ms_(config.interval)
+{
+    // Set up the data type with initial values
+    hello_.index(0);
+    hello_.message("CustomPayloadPool");
+
+    payload_pool_ = std::make_shared<CustomPayloadPool>();
+
+    // Create the participant
+    auto factory = DomainParticipantFactory::get_instance();
+    participant_ = factory->create_participant_with_default_profile(nullptr, StatusMask::none());
```

Comment:
I don't think we should allow creating the participant with profiles.
Maybe we should include a `-d <domain_id> , --domain <domain_id>` option in the CLI parser and use it here:
```suggestion
    DomainParticipantQos p_qos = DEFAULT_PARTICIPANT_QOS;
    factory->get_default_participant_qos(p_qos);
    participant_ = factory->create_participant(config->domain_id, p_qos);
```